### DISCUSSION TITLE
Esp32 build error

### DISCUSSION THREAD

**Discussion by Fnux8890** [esp-rs/espup#421] [Q&A] [answered]
Im trying to use Embassy for an ESP32. As far as I'm aware the Embassy for ESP is based in the esp-rs tools.
When I try and build a simple startup project without any complicated bits just a hello world thing to make sure it works I get this error message

```sh
error: One of the feature flags must be provided: esp32, esp32c2, esp32c3, esp32c6, esp32h2, esp32p4, esp32s2, esp32s3,
   --> /Users/bruger/.cargo/registry/src/index.crates.io-6f17d22bba15001f/esp-hal-0.16.0/build.rs:36:9
    |
36  |           compile_error!(concat!("One of the feature flags must be provided: ", $($all, ", "),*));
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
106 | /     assert_unique_used_features!(
107 | |         "esp32", "esp32c2", "esp32c3", "esp32c6", "esp32h2", "esp32p4", "esp32s2", "esp32s3"
108 | |     );
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `assert_used_features` which comes from the expansion of the macro `assert_unique_used_features` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I'm not sure how to fix this error message.
And not sure if this is an error stemming from the package or something in my project that I'm missing.

**Comment by SergioGasquez** [comment, marked as answer]
You need to specify the chip that you are targeting via a feature, in your case you'd need to use `esp32` feature of `esp-hal`. I'd suggest joining our matrix channel (https://matrix.to/#/#esp-rs:matrix.org) for similar questions since this is not really related with `espup`

**Reply by Fnux8890** [reply]
okay cool.
I'm not sure what matrix is in this regard.

Are there other channels or discussion boards that are more suited for these kinda questions?

**Reply by Fnux8890** [reply]
Just to clarify.
I can see that matrix is a way to chat more generally about questions like this.

I guess what I'm asking is.
Is there some forums or discussion boards that are readily available like stack overflow but closer tied to the esp-rs project so others who have same questions can find it.

**Reply by SergioGasquez** [reply]
> Is there some forums or discussion boards that are readily available like stack overflow but closer tied to the esp-rs project so others who have same questions can find it.

Not really, as a general rule of thumb I would suggest asking in the matrix channel and if that does not solve the question maybe open a discussion in the relevant repo. 

Also, before asking, try searching the in Matrix channel and esp-rs org, just in case a similar question has already been replied.

**Reply by Fnux8890** [reply]
Will do. Thank you very much for the reply 👍
